examples/centralized_telescope.rs

```diff
@@ -0,0 +1,146 @@
+//! Example
+
+// REMOVE!!!!!!!!!!!!!!!!
+#![allow(dead_code)]
+
+mod threshold_signature;
+
+use alba::centralized_telescope::params::Params;
+use alba::centralized_telescope::proof::Proof;
+use alba::centralized_telescope::CentralizedTelescope;
+use rand_chacha::ChaCha20Rng;
+use rand_core::{RngCore, SeedableRng};
+
+use crate::threshold_signature::registration::{ClosedRegistration, RegisteredKeys};
+use crate::threshold_signature::signature::IndividualSignature;
+use crate::threshold_signature::signer::{Candidate, Signer};
+use threshold_signature::aggregate::AggregateSignature;
+
+const DATA_LENGTH: usize = 32;
+pub(crate) type Element = [u8; DATA_LENGTH];
+
+/// Alba proof with aggregate signature
+#[derive(Debug, Clone)]
+pub struct AlbaThresholdProof {
+    /// Aggregate signature
+    pub(crate) aggregate: AggregateSignature,
+    /// Centralized telescope proof
+    pub proof: Proof,
+}
+
+impl AlbaThresholdProof {
+    /// Create an Alba proof
+    /// - Try to aggregate given list of aggregate signatures
+    /// - If aggregation succeeds, create the Alba proof
+    /// - If Alba proof is generated, return the aggregate signature and the Alba proof
+    pub(crate) fn prove<const N: usize>(
+        params: &Params,
+        signatures: &[IndividualSignature],
+        set_size: u64,
+        closed_registration: &ClosedRegistration,
+        msg: &[u8],
+    ) -> Option<Self> {
+        let try_aggregate =
+            AggregateSignature::aggregate::<N>(signatures, closed_registration, msg, set_size);
+        if let Some(aggregate) = try_aggregate {
+            let prover_set: Vec<Element> = aggregate.create_prover_set::<N>();
+            let alba = CentralizedTelescope::create(params);
+            let try_proof = alba.prove(&prover_set);
+            if let Some(proof) = try_proof {
```

Comment:
As you are directly consuming the proof you could do this `if let Some(proof) = alba.prove(&prover_set) {` directly